libvips/foreign/foreign.c

```diff
@@ -1061,6 +1063,10 @@ vips_foreign_load_build( VipsObject *object )
 	if( sequential )
 		load->nocache = TRUE;
 
+	/* For compat, default the deprecated "fail" field to anything.
+	 */
+	load->fail = load->fail_on > VIPS_FAIL_ON_NONE;
```

Comment:
What if someone still sets the deprecated `fail` param to `true`, without setting the new `fail-on` param? Perhaps something like this would work?:
```suggestion
	if( !vips_object_argument_isset( object, "fail" ) )
		load->fail = load->fail_on > VIPS_FAIL_ON_NONE;
	else if( !vips_object_argument_isset( object, "fail-on" ) )
		load->fail_on = load->fail ? VIPS_FAIL_ON_WARNING : VIPS_FAIL_ON_NONE;
```
(untested)

Reply:
Also for compat, perhaps the `load->fail_on > VIPS_FAIL_ON_NONE` line should be `load->fail_on >= VIPS_FAIL_ON_WARNING`?

Reply:
It's extremely confusing :( I kept getting these tests backwards.

`fail_on` controls the sensitivity of loaders to error conditions, from 0 meaning very insensitive, to 3 meaning extremely sensitive indeed. 

`fail_on > none` means any setting other than the default (ie. insensitive). `fail_on > warn` would only enable `fail` if `fail_on` was at the highest sensitivity. I think.

Reply:
... and good spot on `fail` compatibility!

Reply:
No problem! With my last comment I meant that the now deprecated `fail` was at the highest sensitivity (I think). See for example:
https://github.com/libvips/libvips/blob/5cd676a94527c4c1aabdcce534b23d193675fcc5/libvips/foreign/jpeg2vips.c#L811
This was `jpeg->fail` before this PR, hence I thought perhaps we should initialize the compat here:
https://github.com/libvips/libvips/blob/5cd676a94527c4c1aabdcce534b23d193675fcc5/libvips/foreign/foreign.c#L1069
With `load->fail = load->fail_on >= VIPS_FAIL_ON_WARNING` instead. We could also _not_ initialize `load->fail` at all (if we are sure we are not using it anywhere).

Reply:
Oh heh that's true, we can just get rid of `fail` now everything is converted. I'll do that.